src/Manager/ShapeManager.ts

```diff
@@ -0,0 +1,130 @@
+import IShapeInfo from '../Shapes/IShapeInfo';
+import type Shape from '../Shapes/Shape';
+import Point from '../Point';
+import canvas from '../Canvas';
+import ShapeFactory from '../Shapes/ShapeFactory';
+import DraggableShape from '../Shapes/DraggableShape';
+import UpdatableShape from '../Shapes/UpdatableShape';
+import IObserver from '../Observer/IObserver';
+import drawState from '../DrawState';
+import Draw from '../Shapes/Draw';
+import ICanvasEventHandlder from './ICanvasEventHandlder';
+import AbstractObservable from '../Observer/AbstractObservable';
+import IAction from '../Action/IAction';
+import ActionType from '../Action/ActionType';
+import IDocumentEventHandler from './IDocumentEventHandler';
+
+// eslint-disable-next-line max-len
+class ShapeManager extends AbstractObservable<IAction> implements IObserver<IAction>, ICanvasEventHandlder, IDocumentEventHandler {
+  factory : ShapeFactory = new ShapeFactory();
+  shapes : Array<Shape> = [];
+  undoShapes : Array<Shape> = [];
+  currentShape : Shape | null = null;
+  basePoint : Point | null = null;
+  isDrawing = false;
+
+  drawBegin(point : Point) : void {
+    this.isDrawing = true;
+    this.basePoint = point;
+
+    // We dont want to draw anything if DraggableShape
+    if (this.currentShape instanceof DraggableShape) return;
+
+    this.createShape();
+
+    if (!(this.currentShape instanceof UpdatableShape)) {
+      this.currentShape?.draw(this);
+    } else if (this.currentShape !== null) {
+      // Set style for new shape if the shape is not directly draw
```

Comment:
Feels like a repetitive comment

Reply:
Je sais pas, je trouve que ce setStyle sort un peu de null part. Et que donc expliquer que c'est car on va pas draw direct la shape mais l'update, c'est cool.

Reply:
It did seem okay to me but I've worked with this kind of technology in the past, so maybe I'm biased. Let's leave it for newcomers.